src/AvRadio.js

```diff
@@ -0,0 +1,57 @@
+import React, {PropTypes} from 'react';
+import classNames from 'classnames';
+import {Input, FormGroup, Label} from 'reactstrap';
+import AvInput from './AvInput'
+
+let radioPropTypes = Object.assign({}, AvInput.propTypes);
+delete radioPropTypes.name;
+
+export default class AvRadio extends AvInput {
+
+  static contextTypes = Object.assign({}, AvInput.contextTypes, {
+    Group: PropTypes.object.isRequired,
+  });
+
+  static propTypes = radioPropTypes;
+
+  constructor (props) {
+    super(props);
+
+    this.onChangeHandler = ::this.onChangeHandler;
+  }
+
+  onChangeHandler () {
+    this.context.Group.update(this.props.value);
+  }
+
+  render () {
+    const {errorMessage, validate, validationEvent, state, className, tag: Tag, ...attributes} = this.props;
+
+    const classes = classNames(
+      className,
+      this.context.FormCtrl.isTouched[this.props.name] ? 'av-touched' : 'av-untouched',
+      this.context.FormCtrl.isDirty[this.props.name] ? 'av-dirty' : 'av-pristine',
+      this.context.FormCtrl.isBad[this.props.name] ? 'av-bad-input' : null,
```

Comment:
I don't think this is needed. badInput is something the browser tells/sets when the input is something like type="number" which contains a non-number. It's special in that the input's value is empty, but it's not really empty, just invalid (because it's value is empty, it makes it hard to run our own validations, and thus we depend on the browser to let us know). I don't think a radio can have a bad input.